jadx-gui/src/main/java/jadx/gui/search/SearchManager.java

```diff
@@ -0,0 +1,39 @@
+package jadx.gui.search;
+
+import java.util.regex.Pattern;
+
+public class SearchManager {
+	/**
+	 * Generate the good pattern according to the different boolean
+	 *
+	 * @param exp           the searched expression
+	 * @param caseSensitive boolean
+	 * @param wholeWord     boolean
+	 * @param useRegexp     boolean
+	 * @return the pattern
+	 */
+	public static Pattern generatePattern(String exp, boolean caseSensitive, boolean wholeWord, boolean useRegexp) {
+		String word = exp;
```

Comment:
I think it's probably better to use StringBuilder here until you're done processing the string